internal/models/validator.go

```diff
@@ -0,0 +1,31 @@
+package models
+
+import (
+	"context"
+	"errors"
+
+	"github.com/go-playground/validator/v10"
+	"golang.org/x/crypto/ssh"
+)
+
+var validate *validator.Validate
+
+func init() {
+	validate = validator.New()
+
+	_ = validate.RegisterValidation("sshPubkey", func(fl validator.FieldLevel) bool {
+		_, _, _, _, err := ssh.ParseAuthorizedKey([]byte(fl.Field().String()))
+		return err == nil
+	})
+}
+
+func Validate(ctx context.Context, model interface{}) validator.ValidationErrors {
+	err := validate.StructCtx(ctx, model)
+	var validationError validator.ValidationErrors
+	if err != nil && !errors.As(err, &validationError) {
+		// dependency cycle
+		//ctxval.Logger(ctx).Fatal().Msgf("Invalid model passed for validation: %v", model)
```

Comment:
Don't you want to use the logger here?

Reply:
I'd love to, but we've got a depency cycle if we do, I'd like to rething how we approach `ctxval` package, but as of now we can't use logger in models, because ctxval package references with `Account`, which lives in models package and thus dependency cycle.